src/redux/books/books.js

```diff
@@ -1,34 +1,48 @@
+import axios from 'axios';
+
 const ADD_BOOK = 'ADD_BOOK';
 const REMOVE_BOOK = 'REMOVE_BOOK';
-const books = [
-  { id: 1, author: 'Kaweesi Matia', title: 'The World Of Technology' },
-  { id: 2, author: 'Bimal Jalal', title: 'The India Story' },
-  { id: 3, author: 'Ruskin Bond', title: 'Listen to Your Heart: The London Adventure' },
-  { id: 4, author: 'Vinit Karnik ', title: 'Business of Sports: The Winning Formula for Success' },
-];
+const GET_BOOKS = 'GET_BOOKS';
+const books = [];
+
+export const AddBookFunction = (NewBook) => async (dispatch) => {
+  try {
+    await axios.post(`${process.env.REACT_APP_BASE_URL}`, NewBook);
+    return dispatch({ type: ADD_BOOK, payload: NewBook });
+  } catch (err) { return err; }
+};
 
-export const AddBookFunction = (NewBook) => ({
-  type: ADD_BOOK,
-  payload: NewBook,
-});
-export const RemoveBookFunction = (id) => ({
-  type: REMOVE_BOOK,
-  id,
-});
+export const RemoveBookFunction = (id) => async (dispatch) => {
+  try {
+    await axios.delete(`${process.env.REACT_APP_BASE_URL}/${id}`);
+    return dispatch({ type: REMOVE_BOOK, id });
+  } catch (err) { return err; }
+};
+
+export const getBooksFunction = () => async (dispatch) => {
+  try {
+    const response = await axios.get(`${process.env.REACT_APP_BASE_URL}`);
+    return dispatch({ type: GET_BOOKS, payload: response.data });
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connec_to_api.md#:~:text=Use%20fetch%20or%20axios%20with%20createAsyncThunk%20for%20making%20API%20calls.), Please make sure that `createAsyncThunk` is used in the project for creating thunk action creators.